model/config.go

```diff
@@ -61,34 +44,68 @@ func (k Key) String() string {
 // Object references supplied and returned from this interface should be
 // treated as read-only. Modifying them violates thread-safety.
 type ConfigRegistry interface {
-	// Get retrieves a configuration element, bool indicates existence
-	Get(key Key) (proto.Message, bool)
+	// Get retrieves a configuration element.
+	Get(kind, key string) (config proto.Message, exists bool, revision Revision)
 
-	// List returns objects for a kind in a namespace.
-	// Use namespace "" to list resources across namespaces
-	List(kind string, namespace string) (map[Key]proto.Message, error)
+	// List returns objects by type indexed by the key
+	List(kind string) (map[string]proto.Message, error)
```

Comment:
This is a bare minimum. TPR backend only has one level of subkey nesting which is a namespace, and it's not right to use namespace for a subkey. IstioRegistry definitely needs to know how keys are derived and possibly implement subkey listing.